### PR TITLE
Exposing PATCH for Application Security Group (ASG)

### PR DESCRIPTION
**Pull request by rupalivohra**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by EvgenyAgafonchikov** [comment]
LGTM

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json [hunk 1]
```diff
@@ -174,6 +174,56 @@
...
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json]: nit: please make sure this really is a long running opreration

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json [hunk 2]
```diff
@@ -174,6 +174,56 @@
...
+            "description": "Update successful. The operation returns the resulting ApplicationSecurityGroup resource.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationSecurityGroup"
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json]: Please add a `default` section for error status codes

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json (hunk 2 above)
Comment (rupalivohra) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json]: can you share an example of this on an existing PATCH aPI? I was using virtualNetworks PATCH as a reference, didn't see anything like that there.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json (hunk 1 above)
Comment (rupalivohra) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationSecurityGroup.json]: I was using vnet PATCH as an example, this is what it does. We should aim for consistency across our resources, right?

**Comment by dsgouda** [comment]
If these operations are modeled based on existing specs we don't need to change anything here.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**